app/frontend/wallet/shelley/shelley-bitbox02-crypto-provider.ts

```diff
@@ -0,0 +1,350 @@
+    }
+  }
+
+  const derivationScheme = derivationSchemes.v2
+
+  const bb02Network = {
+    [NetworkId.MAINNET]: bitbox02Constants.messages.CardanoNetwork.CardanoMainnet,
+    [NetworkId.TESTNET]: bitbox02Constants.messages.CardanoNetwork.CardanoTestnet,
+  }[network.networkId]
+
+  const version = await withDevice((bitbox02: BitBox02API) => {
+    const version = bitbox02.version().split('.')
+    return Promise.resolve({
+      major: version[0],
+      minor: version[1],
+      patch: version[2],
+    })
+  })
+
+  const getVersion = (): string => `${version.major}.${version.minor}.${version.patch}`
+
+  ensureFeatureIsSupported(CryptoProviderFeature.MINIMAL)
+
+  const getType = () => CryptoProviderType.BITBOX02
+
+  const deriveXpub = CachedDeriveXpubFactory(
+    derivationScheme,
+    config.shouldExportPubKeyBulk && isFeatureSupported(CryptoProviderFeature.BULK_EXPORT),
+    async (derivationPaths: BIP32Path[]) => {
+      return await withDevice(async (bitbox02: BitBox02API) => {
+        const xpubs: Uint8Array[] = await bitbox02.cardanoXPubs(derivationPaths)
+        return xpubs.map(Buffer.from)
+      })
+    },
+    isFeatureSupported(CryptoProviderFeature.BYRON)
+  )
+
+  function isFeatureSupported(feature: CryptoProviderFeature): boolean {
+    switch (feature) {
+      case CryptoProviderFeature.BYRON:
+      case CryptoProviderFeature.POOL_OWNER:
+      case CryptoProviderFeature.MULTI_ASSET:
+      case CryptoProviderFeature.VOTING:
+        return false
+      default:
+        return BITBOX02_VERSIONS[feature]
+          ? hasRequiredVersion(version, BITBOX02_VERSIONS[feature])
+          : hasRequiredVersion(version, BITBOX02_VERSIONS[CryptoProviderFeature.MINIMAL])
+    }
+  }
+
+  function ensureFeatureIsSupported(feature: CryptoProviderFeature): void {
+    if (!isFeatureSupported(feature)) {
+      throw new InternalError(
+        BITBOX02_ERRORS[feature] ?? BITBOX02_ERRORS[CryptoProviderFeature.MINIMAL],
+        {message: `${version.major}.${version.minor}.${version.patch}`}
+      )
+    }
+  }
+
+  function getHdPassphrase(): void {
+    throw new UnexpectedError(UnexpectedErrorReason.UnsupportedOperationError, {
+      message: 'Operation not supported',
+    })
+  }
+
+  function sign(message: HexString, absDerivationPath: BIP32Path): void {
+    throw new UnexpectedError(UnexpectedErrorReason.UnsupportedOperationError, {
+      message: 'Operation not supported',
+    })
+  }
+
+  async function displayAddressForPath(
+    absDerivationPath: BIP32Path,
+    stakingPath: BIP32Path
+  ): Promise<void> {
+    await withDevice(async (bitbox02: BitBox02API) => {
+      await bitbox02.cardanoAddress(bb02Network, {
+        pkhSkh: {
+          keypathPayment: absDerivationPath,
+          keypathStake: stakingPath,
+        },
+      })
+    })
+  }
+
+  function getWalletSecret(): void {
+    throw new UnexpectedError(UnexpectedErrorReason.UnsupportedOperationError, {
+      message: 'Unsupported operation!',
```

Comment:
Inconsistent with `message: 'Operation not supported',` a few lines higher

Reply:
I noticed the same inconsistency in the rest of crypto providers, but I agree it's worth unifying it for good, will do :+1: